stl/inc/xutility

```diff
@@ -2566,10 +2559,7 @@ namespace ranges {
     }
 
     namespace _Rend {
-        template <class _Ty>
-        void rend(_Ty&) = delete;
-        template <class _Ty>
-        void rend(const _Ty&) = delete;
+        void rend(); // Block unqualified name lookup

```

Comment:
is there a reason to write this as `void rend();` instead of `void rend() = delete;`? I don't think we're required to use delete here but it may very well make the compiler diagnostic better

Reply:
`= delete;` is also depicted in the proposal as the expected library implementation, immediately before the Wording section. +1 for your recommendation.

Reply:
`= delete` has no effect on a conforming compiler. These "blocking" overloads always have the wrong number of parameters to match the signatures we're looking for, so they never participate in overload resolution despite blocking unqualified name lookup. This is by design, since it may save a few microseconds of compile time.

I have the habit of not `delete`ing them because there was a two-phase bug in MSVC causing the compiler to reject a dependent function-call when phase one name lookup (unqualified name lookup that happens at template definition time) only finds deleted functions. I _think_ this bug has since been fixed - I'll either `= delete` these or `TRANSITION` them with the pertinent bug number.

Reply:
In that case I don't care so much, since the delete will never improve diagnostics.

Reply:
> I have the habit of not `delete`ing them because there was a two-phase bug in MSVC causing the compiler to reject a dependent function-call when phase one name lookup (unqualified name lookup that happens at template definition time) only finds deleted functions. I _think_ this bug has since been fixed - I'll either `= delete` these or `TRANSITION` them with the pertinent bug number.

Good news: the bug I'm thinking of was fixed. Bad news: the code fails to compile. I was happy not knowing, but now that you've made me look I need to minimize and file a new bug.

Reply:
Pretty much the exact same bug, except in a _require-expression_ (if anyone's curious):
```c++
namespace ranges {
  namespace detail {
    void begin() = delete;

    constexpr bool f0(auto& t) {
      begin(t); // Old bug (now fixed)
      return true;
    }

    template <class T>
    concept ADL = requires(T& t) {
      begin(t); // new bug
    };

    constexpr bool f1(ADL auto&) {
      return true;
    }
  }
}

namespace X {
  struct S {};
  constexpr void begin(S const&) {}
}

constexpr X::S s;
static_assert(ranges::detail::f0(s)); // Fine
static_assert(ranges::detail::f1(s)); // Error
```